keras/src/saving/saving_lib.py

```diff
@@ -91,6 +112,37 @@ def save_model(model, filepath, weights_format="h5"):
         with open(filepath, "wb") as f:
             _save_model_to_fileobj(model, f, weights_format)
 
+def _save_and_upload_model_to_hf(model, hf_path, weights_format):
+    if hf_path.startswith("hf://"):
+        hf_path = hf_path[5:]
+    if hf_path.count("/") > 1:
+        raise ValueError(
+            "Invalid `hf_path` argument: expected `namespace/model_name` format. "
+            f"Received: hf_path={hf_path}"
+        )
+
+    api = huggingface_hub.HfApi(library_name="keras", library_version=keras_version)
+    repo_url = api.create_repo(hf_path, exist_ok=True)
+    repo_id = repo_url.repo_id
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        _save_model_to_folder(model, tmp_dir, weights_format)
+
+        model_card = _MODEL_CARD_TEMPLATE
+
+        if check_pydot():
+            plot_path = os.path.join(tmp_dir, "assets", "config.png")
+            plot_model(model, to_file=plot_path, show_layer_names=True, show_shapes=True, show_dtype=True)
+            model_card += "\n\n![](./assets/config.png)"
```

Comment:
Cool feature! Is `config.png` the right name for the plot?

Also, note that the figure might be potentially be enormous for some models that are complex Functional models, like DenseNet -- would it still render file on the card in that case?

Reply:
> Is config.png the right name for the plot?

I renamed it `summary_plot.png` to be more explicit. What do you think?

>Also, note that the figure might be potentially be enormous for some models that are complex Functional models, like DenseNet -- would it still render file on the card in that case?

Good remark. I'm not sure how to decide whether to render or not the model summary. As a test, I pushed a `keras.applications.DenseNet121` model to the Hub: https://huggingface.co/Wauplin/densenet_121_example. The image loads correctly but it's honestly not really readable/helpful.

Any suggestion how to tackle this? I think that we could either:
1. plot the summary only if number of layers less than a threshold. Mentioning it but I'm not a big fan of this solution as it might be confusing to users.
2. plot the summary and save it to the repo but do not load it in the model card. We can simply add a link to it in the model card and that's it. This way, we don't end up with enormous pages.

Up to you, I can implement either solution!

Reply:
Other idea: instead of loading the summary plot in the model card, would it be possible to generate the `.summary()` text and save it as a markdown table?  Or is this only possible for `Sequential` models?